### PR TITLE
Expose NetDial and NetDialContext

### PR DESCRIPTION
**Pull request by xjtdy888**
We need a custom dialing method, I want to expose it, such as a custom connection for socks5 proxy, or tcp over udp

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -21,6 +23,14 @@ const (
 
 // Config contains various client options.
 type Config struct {
+	// NetDial specifies the dial function for creating TCP connections. If
+	// NetDial is nil, net.Dial is used.
+	NetDial func(network, addr string) (net.Conn, error)
+
+	// NetDialContext specifies the dial function for creating TCP connections. If
+	// NetDialContext is nil, net.DialContext is used.
+	NetDialContext func(ctx context.Context, network, addr string) (net.Conn, error)
```
Comment (FZambia) [inline config.go:32]: Can we only expose NetDialContext?

**Comment by FZambia** [comment]
@xjtdy888 hi, thx for pr, added one review comment.

File: config.go (hunk 1 above)
Comment (xjtdy888) [reply config.go:28]: OK， now only expose NetDialContext

**Comment by FZambia** [comment]
released v0.5.1 with this change

**Pull request merged**